src/WebApp/Pages/Account/Forgot-Password.cshtml.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+using Codidact.Authentication.Domain.Entities;
+using Codidact.Authentication.Infrastructure.Common.Interfaces;
+
+namespace Codidact.Authentication.WebApp.Pages.Account
+{
+    [BindProperties]
+    public class ForgotPasswordModel : PageModel
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IEmailSender<EmailSettings> _emailService;
+
+        public ForgotPasswordModel(
+                          UserManager<ApplicationUser> userManager,
+                          IEmailSender<EmailSettings> emailService
+            )
+        {
+            _userManager = userManager;
+            _emailService = emailService;
+        }
+
+        [Required(ErrorMessage = "E-Mail Address is required")]
+        [DataType(DataType.EmailAddress)]
+        public string Email { get; set; }
+
+        public bool Sent { get; set; }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (ModelState.IsValid)
+            {
+                var user = await _userManager.FindByEmailAsync(Email);
+                if (user != null)
+                {
+                    var token = await _userManager.GeneratePasswordResetTokenAsync(user);
+                    await _emailService.SendResetPassword(Email, token, "http://localhost:8001/");
```

Comment:
- I am bob and forgot my password on the core instance `http://some-codidact-instance.another-site.example`. 

  - This Codidact instance is registered to work with `http://sso.codidact.org`, our authentication instance. 

  - I click the reset password link and get an email like this (simplified):
    ~~~none
    http://sso.codidact.org/account/reset-password?token=xyz
    ~~~

  - I choose a new password and ... `http://sso.codidact.org` does not have any information on which site I wanted to login to.

  - Therefor we need to include a return url parameter in the link:
    ~~~none
    http://sso.codidact.org/account/reset-password?token=xyz&returnurl=...
    ~~~
    This return url would be something like `http://some-codidact-instance.another-site.example/signin` (simplified).

This is already implemented, but currently the return url is hard coded to `http://localhost:8001`. This should be taken from an url parameter. This would be done with the `ReturnUrl` stuff we already have in the login and register pages.